aes-gcm/src/lib.rs

```diff
@@ -129,7 +131,14 @@ pub const C_MAX: u64 = (1 << 36) + 16;
 pub type Nonce<NonceSize> = GenericArray<u8, NonceSize>;
 
 /// AES-GCM tags.
-pub type Tag = GenericArray<u8, U16>;
+pub type Tag<TagSize = U16> = GenericArray<u8, TagSize>;
+
+/// Trait implemented for valid tag sizes, i.e.
+/// [`U12`][consts::U12], [`U13`][consts::U13], [`U14`][consts::U14],
+/// [`U15`][consts::U15] and [`U16`][consts::U16].
+pub trait TagSizeLen: private::SealedTag {}
```

Comment:
To bikeshed some names:

```suggestion
pub trait TagSize: private::SealedTagSize {}
```